tracking-migration/src/main/java/io/customer/tracking/migration/util/JsonAdapter.kt

```diff
@@ -11,4 +22,101 @@ class JsonAdapter {
     fun fromJsonToListOrNull(json: String): JSONArray? = runCatching {
         return JSONArray(json)
     }.getOrNull()
+
+    /**
+     * Parses the migration task from the queue task JSON object and returns the result.
+     * If the task is invalid, it will return failure result with an error message.
+     * If the task is valid, it will return success result with the parsed migration task
+     * that can be processed by the migration processor.
+     */
+    fun parseMigrationTask(task: JSONObject): Result<MigrationTask> = runCatching {
+        val data = task.stringOrNull("data")
+        require(!data.isNullOrBlank()) { "Queue task data is null or blank for $task. Could not run task." }
+
+        val type = task.stringOrNull("type")?.let { enumValueOfOrNull<QueueTaskType>(it) }
+        requireNotNull(type) { "Queue task type is invalid for $task. Could not run task." }
+
+        val taskJson = runCatching { JSONObject(data) }.getOrElse {
+            throw RuntimeException("Queue task data is invalid for $task. Could not run task.")
+        }
+        val timestamp = taskJson.longOrNull("timestamp") ?: Date().time
+
+        val result = when (type) {
```

Comment:
All field names added here for parsing are taken from deleted models to make sure we parse JSON correctly.